lib/modules/apostrophe-pieces/lib/api.js

```diff
@@ -124,6 +124,45 @@ module.exports = function(self, options) {
     return next();
   };
 
+  // This method returns list of suggestions based on highSearchWords array,
+  // it uses the autocomplete cursor to request documents, it also checks for filters.
+  self.getSuggestions = async function(req, term, filters) {
+    const search = self.apos.launder.string(term).toLowerCase();
+    const cursor = getCursorWithFilters(filters);
+    const images = await cursor.autocomplete(search).limit(15).toArray();
+
+    const suggestions = images.reduce((acc, img) => {
+      if (acc.length >= 10) {
+        return acc;
+      }
+
+      const matchingWords = img.highSearchWords.filter((word) => {
+        if (acc.includes(word)) {
+          return false;
+        }
+        return acc.includes(word)
+          ? false
+          : word.startsWith(search);
+      });
+
+      return [
+        ...acc,
+        ...matchingWords
+      ];
+    }, []);
+
+    return suggestions;
+
+    // adding filter to cursor only if this one exist for the current piece
+    function getCursorWithFilters (filters) {
```

Comment:
This will crash on made-up query strings.

Just call:

`self.find(req, {}).queryToFilters(filters, 'manage')`

To apply those that really exist, are safe for the manage view, and actually have a launder method.

Reply:
Thanks for this one, way cleaner indeed. 👍🏼